internal/pkg/certificates/certificates.go

```diff
@@ -0,0 +1,144 @@
+package certificates
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/kubewarden/kubewarden-controller/internal/pkg/constants"
+	controllerCrypto "github.com/kubewarden/kubewarden-controller/internal/pkg/crypto"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	logger = ctrl.Log.WithName("certificates")
+)
+
+// This function will create and store in secrets, when necessary, the root CA and the certificate used in the
+// controller. If the certificate is create, the process exit to force kubernetes to redeploy the pod with the
+// latest version of the certificates.
+func RenewRootCAIfNedded(ctx context.Context, mgr ctrl.Manager, deploymentsNamespace string, controllerWebhookServiceName string) error {
+	logger.Info("Start to verify certificates")
```

Comment:
This should be a debug statement, plus the message is a bit misleading. We are verifying the root CA, not the certificates